packages/block-library/src/navigation/use-navigation-menu.js

```diff
@@ -22,9 +22,16 @@ export default function useNavigationMenu( navigationMenuId ) {
 			const rawNavigationMenu = navigationMenuId
 				? getEntityRecord( ...navigationMenuSingleArgs )
 				: null;
-			const navigationMenu = navigationMenuId
+			let navigationMenu = navigationMenuId
 				? getEditedEntityRecord( ...navigationMenuSingleArgs )
 				: null;
+
+			// getEditedEntityRecord will return the post regardless of status.
+			// Therefore if the found post is not published then we should ignore it.
+			if ( navigationMenu?.status !== 'publish' ) {
+				navigationMenu = null;
+			}
```

Comment:
Navigation menus can be drafts too, that's an important part of the pattern insertion flow. Just wondering what the reasoning behind this change was?

Reply:
Related - https://github.com/WordPress/gutenberg/issues/37354